BinaryEncoding.md

```diff
@@ -234,6 +233,52 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section is not reported and is treated as the section
+being absent. The expectation is that, when a binary WebAssembly module is
+viewed in a browser or other development environment, the names in this section
+will be used as the names of functions in the [text format](TextFormat.md).
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `name*` | sequence of names |
+
+The number of names is determined by the number of declared functions. The
+sequence of names assigns a name to each function index. A name is:
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| size | varuint32 | string length, in bytes |
+| string | bytes | valid utf8 encoding |
+
+### Local Names section
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section is not reported and is treated as the section
+being absent. The expectation is that, when a binary WebAssembly module is
+viewed in a browser or other development environment, the names in this section
+will be used as the names of locals in the [text format](TextFormat.md).
+
+ID: `local_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `name*` | sequence of names |
+
+The number of names is determined by the sum of all functions' number of locals.
```

Comment:
Yes, previously I broke it up b/c locals are too big.  BUT, if we do lengths anyway and then fuse the two lists together so that each function has a name and list of locals (that can be less than the actual number of locals) that solves it all nicely.  I think I'll do that.